src/server/game/Entities/Player/Player.cpp

```diff
@@ -26709,9 +26709,15 @@ void Player::ActivateSpec(uint8 spec)
 
     // xinef: apply glyphs from second spec
     for (uint8 slot = 0; slot < MAX_GLYPH_SLOT_INDEX; ++slot)
-        if (uint32 glyphId = m_Glyphs[GetActiveSpec()][slot])
-            if (GlyphPropertiesEntry const* glyphEntry = sGlyphPropertiesStore.LookupEntry(glyphId))
-                CastSpell(this, glyphEntry->SpellId, TriggerCastFlags(TRIGGERED_FULL_MASK&~(TRIGGERED_IGNORE_SHAPESHIFT|TRIGGERED_IGNORE_CASTER_AURASTATE)));
+    {
+	    uint32 glyphId = m_Glyphs[GetActiveSpec()][slot];
+
+	    if (glyphId)
+		    if (GlyphPropertiesEntry const* glyphEntry = sGlyphPropertiesStore.LookupEntry(glyphId))
+			    CastSpell(this, glyphEntry->SpellId, TriggerCastFlags(TRIGGERED_FULL_MASK&~(TRIGGERED_IGNORE_SHAPESHIFT | TRIGGERED_IGNORE_CASTER_AURASTATE)));
+
+		SetGlyph(slot, glyphId, true);
```

Comment:
This sets glyph even if glyphId doesn't exist?

Reply:
@Sdyess Is there an instance when the glyph doesn't exist?

Reply:
It has to be outside the if statement or else the issue won't be resolved. Besides the code is from TC.

Reply:
That doesn't even make sense. If the glyphid always exists, it would still always execute that function to set glyph.